dbms/src/Flash/Pipeline/Schedule/TaskQueues/FIFOTaskQueue.cpp

```diff
@@ -39,36 +45,47 @@ bool FIFOTaskQueue::take(TaskPtr & task) noexcept
     return true;
 }
 
-bool FIFOTaskQueue::empty() noexcept
+bool FIFOTaskQueue::empty() const
 {
     std::lock_guard lock(mu);
     return task_queue.empty();
 }
 
-void FIFOTaskQueue::close()
+void FIFOTaskQueue::finish()
 {
     {
         std::lock_guard lock(mu);
-        is_closed = true;
+        is_finished = true;
     }
     cv.notify_all();
 }
 
-void FIFOTaskQueue::submit(TaskPtr && task) noexcept
+void FIFOTaskQueue::submit(TaskPtr && task)
 {
-    assert(task);
+    if unlikely (is_finished)
+    {
+        FINALIZE_TASK(task);
+        return;
+    }
+
     {
         std::lock_guard lock(mu);
+        assert(task);
```

Comment:
```suggestion
        RUNTIME_CHECK(task);
```

Reply:
Since all task_ptr's must be non-nullptr, I've used assert here, maybe delete the useless assert's?

Reply:
removed.